src/Firestore/Query.php

```diff
@@ -629,4 +786,24 @@ private function newQuery(array $additionalConfig, $overrideTopLevelKeys = false
             $query
         );
     }
+
+    /**
+     * Clean up the query array before sending.
+     *
+     * Some optimizations cannot be performed ahead of time and must be done
+     * at execution.
+     *
+     * @param array $query The incoming query
+     * @return array The final query data,
+     */
+    private function finalQueryPrepare(array $query)
+    {
+        if (isset($query['where']['compositeFilter']) && count($query['where']['compositeFilter']['filters']) === 1) {
+            $filter = $query['where']['compositeFilter']['filters'][0];
+            unset($query['where']);
+            $query['where'] = $filter;
```

Comment:
What happens if I use this Query instance later and add a new filter?

Reply:
The `$query` local in this method is passed by value. Modifications in this method do not propagate back to modify the state of the instance. Re-use of the Query instance will not be affected by this method.